### ISSUE TITLE
Предложение: отказ от файла настроек

### ISSUE THREAD

**Issue by EvilBeaver** [ruimage/Jira-os#18]
Почему бы не убрать файл настроек и не предоставить клиенту библиотеки самому решать - где хранить параметры - в файле, в базе данных, в переменных окружения, получать из аргументов командной строки или жестко в коде прописывать?

Т.е. библиотека должна дать настроечные параметры в виде свойств или методов. А установка этих свойств должна делаться клиентским приложением, которое библиотеку использует.

**Comment by ruimage** [comment]
@EvilBeaver  Есть примеры где посмотреть примеры?:

- из базы - что имеется ввиду
- переменные окружения - тоже не понимаю.

**Comment by EvilBeaver** [comment]
Ну смотрите, вот есть класс УправлениеКонфигуратором из библиотеки v8runner.

В нем указывается - в какую базу 1С надо отправлять команды - "выгрузить CF", "обновить из хранилища" и т.п.

Сама библиотека не знает - как ее будут использовать, она предоставляет интерфейс к типовым операциям с базой 1С.

С JIRA аналогично. Человек, который использует библиотеку JIRA может не захотеть хранить настройки в файле. Например, он пишет систему, где все настройки лежат в базе. Или это вообще одноразовый скрипт и настройки сразу в коде прописаны и создавать служебный файл вообще не планировалось.

Я предлагаю передавать параметры в конструктор класса JIRA, а не хранить их в файле. Хранение и управление настройками это прерогатива использующего приложения, а библиотека должна предоставлять только интерфейс.

**Comment by ruimage** [comment]
Тогда я понимаю ишшуз так:
Добавить функционал:
 - При создании экземпляра класса Jira настройки подключения устанавливать в конструктор
Удалить функционал
- считывание настроек из файла json по имени

ОК?

**Comment by EvilBeaver** [comment]
Да, файл настроек не нужен в библиотеке. В библиотеке все настройки должны быть свойствами или параметрами методов/конструкторов.

**Comment by EvilBeaver** [comment]
up
